modules/core/test/test_utils.cpp

```diff
@@ -0,0 +1,179 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+
+using namespace cv;
+
+namespace {
+
+static const char * const keys =
+    "{ h help    |       | print help }"
+    "{ i info    | false | print info }"
+    "{ t true    | true  | true value }"
+    "{ n unused  |       | dummy }"
+;
+
+TEST(CommandLineParser, testHas_noValues)
+{
+    const char* argv[] = {"<bin>", "-h", "--info"};
+    const int argc = 3;
+    cv::CommandLineParser parser(argc, argv, keys);
+    EXPECT_TRUE(parser.has("help"));
+    EXPECT_TRUE(parser.has("h"));
+    EXPECT_TRUE(parser.has("info"));
+    EXPECT_TRUE(parser.has("i"));
+    EXPECT_FALSE(parser.has("n"));
+    EXPECT_FALSE(parser.has("unused"));
+    EXPECT_FALSE(parser.has("q")); // TODO Throw ???
+}
+TEST(CommandLineParser, testHas_TrueValues)
+{
+    const char* argv[] = {"<bin>", "-h=TRUE", "--info=true"};
+    const int argc = 3;
+    cv::CommandLineParser parser(argc, argv, keys);
+    EXPECT_TRUE(parser.has("help"));
+    EXPECT_TRUE(parser.has("h"));
+    EXPECT_TRUE(parser.has("info"));
+    EXPECT_TRUE(parser.has("i"));
+    EXPECT_FALSE(parser.has("n"));
+    EXPECT_FALSE(parser.has("unused"));
+    EXPECT_FALSE(parser.has("q")); // TODO Throw ???
```

Comment:
good idea. asking for a undeclared key is likely a programming error. (e.g. typo in key name)